move proposals and proposals details page to page package

Sirmorrison opened the pull request:
Resolve #533 

* This pr moved the proposals page, proposals details and proposal vote modal files to the proposal package.

Sirmorrison committed:
move proposals and proposals details page to page package

Sirmorrison committed:
move proposals page and proposals details page to proposal package

Sirmorrison committed:
move proposal vote modal to proposal package

beansgum commented on ui/main_page.go:
```diff
@@ -34,6 +35,7 @@ type mainPage struct {
 	current, previous string
 	pages             map[string]Page
 	sendPage          *send.Page // reuse value to keep data persistent onresume.
+	// proposalPage *proposal.Page
```
Remove

beansgum commented on ui/page/components/components.go:
```diff
@@ -5,19 +5,22 @@ package components
 	"gioui.org/text"
 	"gioui.org/unit"
+	"golang.org/x/text/message"
+
```
Organise imports.

beansgum commented on ui/page/components/log.go:
```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2017, The dcrdata developers
```
Not required, the components package does not need its own logger.

beansgum commented on ui/page/proposal/log.go:
```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2017, The dcrdata developers
+// See LICENSE for details.
+
+package proposal
```
Not required, the proposal package does not need its own logger.

beansgum commented on ui/page/proposal/page.go:
```diff
@@ -0,0 +1,200 @@
+package proposal
```
It's misleading naming this file as `Page` because this is not the only page in this package, I can't tell if this is for the proposal details or list without looking into the code, can keep the name as "Proposals".

beansgum commented on ui/page/proposal/proposal_details_page.go:
```diff
@@ -13,23 +13,25 @@ import (
 type proposalDetails struct {
-	theme              *decredmaterial.Theme
+	*load.Load
+	wallet *wallet.Wallet
```
This struct is redundant and should not be used.

Sirmorrison replied:
Which of the struct is redundant? `proposalDetails` or `proposalItemWidgets`

beansgum replied:
`wallet.Wallet` Is redundant.

beansgum commented on ui/page/proposal/proposal_details_page.go:
```diff
@@ -382,7 +386,7 @@ func (pg *proposalDetails) Layout(gtx C) D {
 			} else {
 				var err error
-				proposalDescription, err = common.wallet.FetchProposalDescription(proposal.Token)
+				proposalDescription, err = pg.wallet.FetchProposalDescription(proposal.Token)
```
```suggestion
				proposalDescription, err = pg.WL.MultiWallet.Politeia.FetchProposalDescription(proposal.Token)
```

Sirmorrison committed:
remove proposal and componet package log files
- set propsals return page, rename proposal page.go file to proposals_page
- clean up code

Sirmorrison committed:
fix lint error
